compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/builders/ExpressionHelpers.kt

```diff
@@ -40,6 +40,9 @@ inline fun IrBuilderWithScope.irLetS(
     nameHint: String? = null,
     body: (IrValueSymbol) -> IrExpression
 ): IrExpression {
+    if (value is IrGetValue)
```

Comment:
@sfs Is this optimization still required?
As I see this method only used in DataClassMembersGenerator and a getter call is always passed

Reply:
You're right, it's no longer required. I would still like to leave it in, since it will probably come up again in the future (e.g., when generating a safe call in the IR). What do you think?

Reply:
We need some ir-text test here, but it's not clear how test anything in coming up changes. Maybe best is to move this commit to separate pr for now and discuss details there

Reply:
I've removed it from this PR. I will try to commit it as part of #2362, where the optimization is actually required for reasonable codegen.